osu.Game/Rulesets/Scoring/JudgementProcessor.cs

```diff
@@ -151,6 +151,8 @@ protected void SimulateAutoplay(IBeatmap beatmap)
             {
                 var judgement = obj.CreateJudgement();
 
+                HitResultExtensions.ValidateHitResultPair(judgement.MaxResult, judgement.MinResult);
```

Comment:
I'm not sure this is a particularly good place to assert this, though. I'd want it closer to the actual construction of the judgement itself so that it fails as early as possible.

Also there's an argument here that this method throwing is overkill and it should stop at asserting. Dunno about that.

Reply:
> The reason I didn't put this in `Judgement`'s constructor is because someone could theoretically change its value at run time.

That would be vile, but you do have a point...

> We already do some validation in `DHO.ApplyResult()`, maybe it could go alongside it?

That would probably be the other best place to put it, yeah. I did have an alarm bell ringing that doing so may cause a performance hit, but I don't think it's gonna be significant anywhere outside of like frame-stable replay catchup maybe.